src/finance/MainnetSwapSteward.sol

```diff
@@ -0,0 +1,189 @@
+ * the Collector of funds by doing multiple swaps from A -> B -> A with slippage, where the slippage ends up draining the Collector.
+ * The budget can only be updated by the DAO via a Governance vote.
+ *
+ * -- Permissions
+ * The contract implements OwnableWithGuardian.
+ * The owner will always be the respective network Short Executor (governance).
+ * The guardian role will be given to a Financial Service provider of the DAO.
+ *
+ * -- Swap Pairs Information
+ *
+ * The swaps executed by this contract will be performed at the discretion of the finance stewards in order to
+ * achieve different matters on behalf of the DAO.
+ *
+ * GHO Price Stability: Swap stables for GHO in order to regain the price objective
+ * Merit, Streams and similar: Swap stables for GHO in order to have funding for these activities
+ * Long-tail asset exposure reduction: Swap long-tail assets to stablecoins or WETH in order to reduce exposure and
+ * or to consolidate the DAO's holdings
+ * LSTs: Swap underlying tokens into LSTs in order to generate yield for the DAO
+ * Majors: Swap wBTC to wETH as part of the treasury strategy
+ *
+ */
+contract MainnetSwapSteward is IMainnetSwapSteward, OwnableWithGuardian, Multicall {
+  using CU for ICollector;
+  using CU for CU.SwapInput;
+
+  /// @inheritdoc IMainnetSwapSteward
+  uint256 public constant MAX_SLIPPAGE = 1000; // 10%
+
+  /// @inheritdoc IMainnetSwapSteward
+  AaveSwapper public immutable SWAPPER;
+
+  /// @inheritdoc IMainnetSwapSteward
+  ICollector public immutable COLLECTOR;
+
+  /// @inheritdoc IMainnetSwapSteward
+  address public milkman;
+
+  /// @inheritdoc IMainnetSwapSteward
+  address public priceChecker;
+
+  /// @inheritdoc IMainnetSwapSteward
+  mapping(address fromToken => mapping(address toToken => bool isApproved)) public swapApprovedToken;
+
+  /// @inheritdoc IMainnetSwapSteward
+  mapping(address token => address oracle) public priceOracle;
+
+  /// @inheritdoc IMainnetSwapSteward
+  mapping(address token => uint256 budget) public tokenBudget;
+
+  constructor(
+    address initialOwner,
+    address initialGuardian,
+    address collector,
+    address swapper,
+    address initialMilkman,
+    address initialPriceChecker
+  ) OwnableWithGuardian(initialOwner, initialGuardian) {
+    _setMilkman(initialMilkman);
+    _setPriceChecker(initialPriceChecker);
+
+    COLLECTOR = ICollector(collector);
+    SWAPPER = AaveSwapper(swapper);
+  }
+
+  /// @inheritdoc IMainnetSwapSteward
+  function tokenSwap(address fromToken, address toToken, uint256 amount, uint256 slippage) external onlyOwnerOrGuardian {
+    address fromOracle = priceOracle[fromToken];
+    address toOracle = priceOracle[toToken];
+
+    _validateSwap(fromToken, toToken, fromOracle, toOracle, amount, slippage);
+
+    tokenBudget[fromToken] -= amount;
+
+    COLLECTOR.swap(
+      address(SWAPPER), CU.SwapInput(milkman, priceChecker, fromToken, toToken, fromOracle, toOracle, amount, slippage)
```

Comment:
when using so many params, better to use {} object format to ensure they are passed in correct order.